tests/math/test_is_independent.py

```diff
@@ -234,337 +222,331 @@ def lin(x, weights=None):
         assert len(recwarn) == 0
 
 
-if have_jax:
-
-    class TestIsIndependentJax:
-        """Tests for is_independent, which tests a function to be
-        independent of its inputs, using JAX."""
-
-        interface = "jax"
-
-        @pytest.mark.parametrize("num", [0, 1, 2])
-        @pytest.mark.parametrize(
-            "args",
-            [
-                (0.2,),
-                (1.1, 3.2, 0.2),
-                (np.array([[0, 9.2], [-1.2, 3.2]]),),
-                (0.3, [1, 4, 2], np.array([0.3, 9.1])),
-            ],
-        )
-        @pytest.mark.parametrize("bounds", [(-1, 1), (0.1, 1.0211)])
-        def test_get_random_args(self, args, num, bounds):
-            """Tests the utility ``_get_random_args`` using a fixed seed."""
-            seed = 921
-            rnd_args = _get_random_args(args, self.interface, num, seed, bounds)
-            assert len(rnd_args) == num
-            np.random.seed(seed)
-            for _rnd_args in rnd_args:
-                expected = tuple(
-                    np.random.random(np.shape(arg)) * (bounds[1] - bounds[0]) + bounds[0]
-                    for arg in args
-                )
-                assert all(np.allclose(_exp, _rnd) for _exp, _rnd in zip(expected, _rnd_args))
-
-        dev = qml.device("default.qubit", wires=1)
-
-        @qml.qnode(dev, interface=interface)
-        def const_circuit(x, y):
-            qml.RX(0.1, wires=0)
-            return qml.expval(qml.PauliZ(0))
-
-        constant_functions = [
-            const_circuit,
-            lambda x: np.arange(20).reshape((2, 5, 2)),
-            lambda x: (np.ones(3), -0.1),
-            jax.jacobian(lambda x, y: 4.0 * x - 2.1 * y, argnums=[0, 1]),
-        ]
-
-        args_constant = [
-            (0.1, np.array([-2.1, 0.1])),
-            (1.2,),
-            (np.ones((2, 3)),),
-            (jax.numpy.ones((3, 8)) * 0.1, -0.2 * jax.numpy.ones((3, 8))),
-        ]
-
-        @qml.qnode(dev, interface=interface)
-        def dependent_circuit(x, y, z):
-            qml.RX(0.1, wires=0)
-            qml.RY(y / 2, wires=0)
-            qml.RZ(qml.math.sin(z), wires=0)
-            return qml.expval(qml.PauliX(0))
-
-        dependent_functions = [
-            dependent_circuit,
-            jax.numpy.array,
-            lambda x: (1 + qml.math.tanh(1000 * x)) / 2,
-            *dependent_lambdas,
-        ]
-
-        args_dependent = [
-            (0.1, np.array(-2.1), -0.9),
-            (-4.1,),
-            (jax.numpy.ones((3, 8)) * 1.1,),
-            *args_dependent_lambdas,
-        ]
-
-        @pytest.mark.parametrize("func, args", zip(constant_functions, args_constant))
-        def test_independent(self, func, args):
-            """Tests that an independent function is correctly detected as such."""
-            assert is_independent(func, self.interface, args)
+class TestIsIndependentJax:
```

Comment:
Seems like we could instead mark inidividual classes with `pytest.mark.jax` , `pytest.mark.tf`, etc. Instead of marking the whole file with `pytest.mark.math`.

Reply:
You are right I will update it!

Reply:
In fact it is not possible because we need the interfaces to be imported at the file level because inputs have types `tf`, `torch`, `jax`. Therefore I kept it in the `all_interfaces` marker